src/lib/useAuthState.ts

```diff
@@ -1,14 +1,19 @@
 /* eslint-disable import/prefer-default-export */
-import { getKeys } from "@near-js/biometric-ed25519/lib";
+import { getKeys } from '@near-js/biometric-ed25519/lib';
 import { KeyPairEd25519 } from '@near-js/crypto';
-import { useEffect, useMemo, useState } from "react"
-import { useSearchParams } from "react-router-dom/dist";
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom/dist';
 
-import FastAuthController from "./controller";
-import { network, networkId } from "../utils/config";
+import FastAuthController from './controller';
+import { network, networkId } from '../utils/config';
 
-export const useAuthState = () => {
-  const [authenticated, setAuthenticated] = useState(undefined);
```

Comment:
This needs to start as `undefined` because we need to identify when the user has not authenticated yet vs failed authentication in some cases

Reply:
On that exact reason, I used `controllerState` which already does it inside the code and export it:

https://github.com/near/fast-auth-signer/pull/51/files#diff-e32098d15337091eba6e38cf673f428139718632e81710d1e180d58836cae11bR12?plain=1

Reply:
Eg.

https://github.com/near/fast-auth-signer/pull/51/files#diff-8e391524757018ec63bd655f87614c8e65b6457ff6b5150c710bd026796da112R111

https://github.com/near/fast-auth-signer/pull/51/files#diff-8e391524757018ec63bd655f87614c8e65b6457ff6b5150c710bd026796da112R125

Reply:
so I have updated the logic on Sign component

Reply:
`controllerState` could be true (key is retrieved from keyStore) while the user still has not authenticated yet right (e.g. they are still selecting an account from the list). I don't mind exposing `controllerState` from this hook but I prefer for `authenticated` to have a loading state as well and for components to use that to determine if the user has fully authenticated instead of `controllerState` since `controllerState` is one of the variables that determines whether `authenticated` is `true` or not